src/lib/storage/proposal/partition_creator.rb

```diff
@@ -87,9 +87,15 @@ def total_free_size
 
         # List of free spaces in the devicegraph
         #
+        # @param TODO: documentation
         # @return [FreeDiskSpacesList]
-        def free_spaces
-          candidate_disks.free_disk_spaces.with do |space|
+        def free_spaces(disk_name = nil)
```

Comment:
This line killed me:

> free_spaces(vol.disk).first

because (without having looked up the implementation of free_spaces) my thought was: Isn't vol.disk usually nil? That's certainly wrong code...

Maybe you have hidden the binding to certain disks a bit too low-level.

But that's just my thoughts. I'm fine with the code as it is.